tests/unittests/test_request_timeout.py

```diff
@@ -0,0 +1,158 @@
+from tap_google_sheets.client import GoogleClient
+import unittest
+from unittest import mock
+from unittest.case import TestCase
+from requests.exceptions import Timeout, ConnectionError
+
+class TestBackoffError(unittest.TestCase):
+    '''
+    Test that backoff logic works properly.
+    '''
+    @mock.patch('tap_google_sheets.client.requests.Session.request')
+    @mock.patch('tap_google_sheets.client.GoogleClient.get_access_token')
+    def test_request_timeout_and_backoff(self, mock_get_token, mock_request):
+        """
+        Check whether the request backoffs properly for request() for 5 times in case of Timeout error.
+        """
+        mock_request.side_effect = Timeout
+        with self.assertRaises(Timeout):
+            client = GoogleClient("dummy_client_id", "dummy_client_secret", "dummy_refresh_token", 300)
+            client.request("GET")
```

Comment:
I think the client instantiation should be moved outside of this with block. It is unclear what is being tested otherwise.

Reply:
Yes, removed it from the `with`